### PR TITLE
Add APIVersion and APIVersionSetID to import

### LINKED ISSUE #4985: Cannot create multiple versioned APIs with Swagger
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform:  v0.12.13
Azure RM provider : v1.37.0

### Affected Resource(s)

* `azurerm_api_management_api`

### Terraform Configuration Files

```hcl
resource "azurerm_api_management_api_version_set" "client-api-versionset" {
  count               = length(var.client_services)
  name                = "${var.client_short_name}-${var.environment_name}-${var.client_services[count.index]}"
  resource_group_name = var.apim_resource_group_name
  api_management_name = "${data.azurerm_api_management.apim.name}"
  display_name        = "${var.client_short_name}-${var.environment_name}-${var.client_services[count.index]}"
  versioning_scheme   = "Segment"
}


resource "azurerm_api_management_api" "client-api-v1" {
  count               = length(var.client_services)
  name                = "${var.client_short_name}-${var.environment_name}-${var.client_services[count.index]}-v1"
  resource_group_name = var.apim_resource_group_name
  api_management_name = data.azurerm_api_management.apim.name
  revision            = var.api_version
  display_name        = "${var.client_short_name}-${var.environment_name}-${var.client_services[count.index]}-v1"
  path                = "${var.client_short_name}/${var.environment_name}/${var.client_services[count.index]}"
  protocols           = ["https"]
  service_url         = "https://${var.client_short_name}-${var.environment_name}.client.contoso.com/${var.client_services[count.index]}/dpo"
  version             = "v1"
  version_set_id      = azurerm_api_management_api_version_set.client-api-versionset[count.index].id
  import {
    content_format = "swagger-json"
    content_value  = templatefile("${path.module}/open-api-definition/client-v1-swagger.tmpl", { api_name = "${var.client_short_name}-${var.environment_name}-${var.client_services[count.index]}-v1" })

  }
}


resource "azurerm_api_management_api" "client-api-v2" {
  count               = length(var.client_services)
  name                = "${var.client_short_name}-${var.environment_name}-${var.client_services[count.index]}-v2"
  resource_group_name = var.apim_resource_group_name
  api_management_name = "${data.azurerm_api_management.apim.name}"
  revision            = var.api_version
  display_name        = "${var.client_short_name}-${var.environment_name}-${var.client_services[count.index]}-v2"
  path                = "${var.client_short_name}/${var.environment_name}/${var.client_services[count.index]}"
  protocols           = ["https"]
  service_url         = "https://${var.client_short_name}-${var.environment_name}.client.contoso.com/${var.client_services[count.index]}/dpo"
  version             = "v2"
  version_set_id      = azurerm_api_management_api_version_set.client-api-versionset[count.index].id
  import {
    content_format = "swagger-json"
    content_value  = templatefile("${path.module}/open-api-definition/client-v2-swagger.tmpl", { api_name = "${var.client_short_name}-${var.environment_name}-${var.client_services[count.index]}-v2" })

  }
}
```

### Expected Behavior

An API with two versions should be created

### Actual Behavior

The version set and first API (v1) is created fine, the second version (v2) fails with this error:

```
Error: Error creating/updating API Management API "sct001-test-dpo2-v2" (Resource Group "CLIENT"): apimanagement.APIClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="ValidationError" Message="One or more fields contain incorrect values:" Details=[{"code":"ValidationError","message":"Cannot create API 'sct001-test-dpo2-v2' with the same Path 'sct001/test/dpo2' as API 'sct001-test-dpo2-v1'  unless it's a part of the same version set","target":"path"}]
```
If I create the APIs without importing a swagger file then it works as expected and I get two empty versioned APIs

### Steps to Reproduce

1. Create two swagger files for versions of the API
2. Create a version set
3. Create two apis, using the version set and swagger files
4. Terraform apply

### References

* #0000

### PR DESCRIPTION
**Pull request by sam-cogan**
Adds the APIVersion and APIVersionSetID to the REST call for creating an API using an import from a file (swagger, WSDL etc.). This fixes the bug in https://github.com/terraform-providers/terraform-provider-azurerm/issues/4985 that would result in creation failing if attempting to create more than 1 version of a single API with the same path.

Fixes #4985

### CODE REVIEW

File: azurerm/resource_arm_api_management_api.go [hunk 1]
```diff
@@ -253,9 +260,6 @@ func resourceArmApiManagementApiCreateUpdate(d *schema.ResourceData, meta interf
 	displayName := d.Get("display_name").(string)
 	serviceUrl := d.Get("service_url").(string)
 
-	version := d.Get("version").(string)
-	versionSetId := d.Get("version_set_id").(string)
-
 	if version != "" && versionSetId == "" {
 		return fmt.Errorf("Error setting `version` without the required `version_set_id`")
```
[the remark this answers is not in the document]
Comment (sam-cogan) [reply azurerm/resource_arm_api_management_api.go]: Thanks, have moved the validation to earlier in the code to sit along side the value declaration.

Not sure what missing links your referring to?

**Comment by sam-cogan** [comment]
@tombuildsstuff I've updated the code to move the validation earlier, but not sure what you meant by "adding back missing links"

**Comment by sam-cogan** [comment]
@tombuildsstuff is there anything left that needs doing to get this merged?

**Comment by tombuildsstuff** [comment]
chatting offline: this is fine, I'd confused the comment when writing the PR summary 👍

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2019-12-12 at 15 28 09

**Pull request merged**